test/validations-test.js

```diff
@@ -0,0 +1,46 @@
+const chai = require("chai");
+const sinonChai = require("sinon-chai");
+const expect = chai.expect;
+const Liquid = require("../index.js");
+chai.use(sinonChai);
+
+describe("check whether variable used in liquidjs expressions has been defined before usage", () => {
+  const commonSetup = () => {
+    const engine = Liquid({});
+    return engine;
+  };
+  it("should return an empty array if all variables being used in computations have been defined earlier before being used", () => {
```

Comment:
Need more tests
1. "should fail if variable used is not defined before"
2. "should fail if variable used inside IF condition is not defined before"
3. "should fail if variable used inside ELSE condition is not defined before"
4. "should not fail if variable used inside nested IF condition is defined in previous IF layer"
etc